Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
Well but the problem still remains if I delete the object then will these function make the deletion from the processing?